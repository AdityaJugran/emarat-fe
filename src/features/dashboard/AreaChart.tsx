import {
	AreaChart as AreaChartComponent,
	Area,
	XAxis,
	YAxis,
	CartesianGrid,
	Tooltip,
} from 'recharts';
import { useOrientation } from 'config/hooks';
import StyledTitle from 'features/shared/components/styledComponents/StyledTitle';

type AreaChartProps = {
	data: {
		month: string;
		amount: number;
	}[];
};

export default function AreaChart({ data }: AreaChartProps) {
	const { isMobileSize } = useOrientation();
	const slicedData = isMobileSize ? data.slice(0, 5) : data.slice(0, 6);
	const width = Math.max(data.length * 60, 500);
	return (
		<div className="sm:pb-6">
			<StyledTitle className="mb-8">Monthly Revenue</StyledTitle>
			<AreaChartComponent
				width={isMobileSize ? 320 : width}
				height={isMobileSize ? 200 : 250}
				data={slicedData}
			>
				<CartesianGrid strokeDasharray="3 3" />
				<XAxis dataKey="month" />
				<YAxis className="text-xs" />
				<Tooltip />
				<Area
					type="monotone"
					dataKey="amount"
					stroke="#8884d8"
					fill="#8884d8"
				/>
			</AreaChartComponent>
		</div>
	);
}
