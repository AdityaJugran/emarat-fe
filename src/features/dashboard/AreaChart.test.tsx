import AreaChart from 'features/dashboard/AreaChart';
import renderer from 'react-test-renderer';

it('Amenity Skeleton', () => {
	const areaChartData = {
		month: 'jan',
		amount: 200,
	};
	const tree = renderer.create(<AreaChart data={[areaChartData]} />).toJSON();
	expect(tree).toMatchSnapshot();
});
