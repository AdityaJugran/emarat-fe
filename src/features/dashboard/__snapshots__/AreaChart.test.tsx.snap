// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Amenity Skeleton 1`] = `
<div
  className="sm:pb-6"
>
  <p
    className="sc-bdfBQB bbCVRC mb-8"
  >
    Monthly Revenue
  </p>
  <div
    className="recharts-wrapper"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "cursor": "default",
        "height": 250,
        "position": "relative",
        "width": 500,
      }
    }
  >
    <svg
      className="recharts-surface"
      height={250}
      version="1.1"
      viewBox="0 0 500 250"
      width={500}
    >
      <defs>
        <clipPath
          id="recharts1-clip"
        >
          <rect
            height={210}
            width={430}
            x={65}
            y={5}
          />
        </clipPath>
      </defs>
      <g
        className="recharts-cartesian-grid"
      >
        <g
          className="recharts-cartesian-grid-horizontal"
        >
          <line
            fill="none"
            height={210}
            offset={
              Object {
                "bottom": 35,
                "brushBottom": 35,
                "height": 210,
                "left": 65,
                "right": 5,
                "top": 5,
                "width": 430,
              }
            }
            stroke="#ccc"
            strokeDasharray="3 3"
            width={430}
            x={65}
            x1={65}
            x2={495}
            y={5}
            y1={215}
            y2={215}
          />
          <line
            fill="none"
            height={210}
            offset={
              Object {
                "bottom": 35,
                "brushBottom": 35,
                "height": 210,
                "left": 65,
                "right": 5,
                "top": 5,
                "width": 430,
              }
            }
            stroke="#ccc"
            strokeDasharray="3 3"
            width={430}
            x={65}
            x1={65}
            x2={495}
            y={5}
            y1={162.5}
            y2={162.5}
          />
          <line
            fill="none"
            height={210}
            offset={
              Object {
                "bottom": 35,
                "brushBottom": 35,
                "height": 210,
                "left": 65,
                "right": 5,
                "top": 5,
                "width": 430,
              }
            }
            stroke="#ccc"
            strokeDasharray="3 3"
            width={430}
            x={65}
            x1={65}
            x2={495}
            y={5}
            y1={110}
            y2={110}
          />
          <line
            fill="none"
            height={210}
            offset={
              Object {
                "bottom": 35,
                "brushBottom": 35,
                "height": 210,
                "left": 65,
                "right": 5,
                "top": 5,
                "width": 430,
              }
            }
            stroke="#ccc"
            strokeDasharray="3 3"
            width={430}
            x={65}
            x1={65}
            x2={495}
            y={5}
            y1={57.5}
            y2={57.5}
          />
          <line
            fill="none"
            height={210}
            offset={
              Object {
                "bottom": 35,
                "brushBottom": 35,
                "height": 210,
                "left": 65,
                "right": 5,
                "top": 5,
                "width": 430,
              }
            }
            stroke="#ccc"
            strokeDasharray="3 3"
            width={430}
            x={65}
            x1={65}
            x2={495}
            y={5}
            y1={5}
            y2={5}
          />
        </g>
        <g
          className="recharts-cartesian-grid-vertical"
        >
          <line
            fill="none"
            height={210}
            offset={
              Object {
                "bottom": 35,
                "brushBottom": 35,
                "height": 210,
                "left": 65,
                "right": 5,
                "top": 5,
                "width": 430,
              }
            }
            stroke="#ccc"
            strokeDasharray="3 3"
            width={430}
            x={65}
            x1={280}
            x2={280}
            y={5}
            y1={5}
            y2={215}
          />
          <line
            fill="none"
            height={210}
            offset={
              Object {
                "bottom": 35,
                "brushBottom": 35,
                "height": 210,
                "left": 65,
                "right": 5,
                "top": 5,
                "width": 430,
              }
            }
            stroke="#ccc"
            strokeDasharray="3 3"
            width={430}
            x={65}
            x1={65}
            x2={65}
            y={5}
            y1={5}
            y2={215}
          />
          <line
            fill="none"
            height={210}
            offset={
              Object {
                "bottom": 35,
                "brushBottom": 35,
                "height": 210,
                "left": 65,
                "right": 5,
                "top": 5,
                "width": 430,
              }
            }
            stroke="#ccc"
            strokeDasharray="3 3"
            width={430}
            x={65}
            x1={495}
            x2={495}
            y={5}
            y1={5}
            y2={215}
          />
        </g>
      </g>
      <g
        className="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
      >
        <line
          className="recharts-cartesian-axis-line"
          fill="none"
          height={30}
          orientation="bottom"
          stroke="#666"
          type="category"
          width={430}
          x={65}
          x1={65}
          x2={495}
          y={215}
          y1={215}
          y2={215}
        />
        <g
          className="recharts-cartesian-axis-ticks"
        >
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={30}
              orientation="bottom"
              stroke="#666"
              type="category"
              width={430}
              x={65}
              x1={280}
              x2={280}
              y={215}
              y1={221}
              y2={215}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={30}
              orientation="bottom"
              stroke="none"
              textAnchor="middle"
              type="category"
              width={430}
              x={280}
              y={223}
            >
              <tspan
                dy="0.71em"
                x={280}
              >
                jan
              </tspan>
            </text>
          </g>
        </g>
      </g>
      <g
        className="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
      >
        <line
          className="recharts-cartesian-axis-line"
          fill="none"
          height={210}
          orientation="left"
          stroke="#666"
          type="number"
          width={60}
          x={5}
          x1={65}
          x2={65}
          y={5}
          y1={5}
          y2={215}
        />
        <g
          className="recharts-cartesian-axis-ticks"
        >
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={210}
              orientation="left"
              stroke="#666"
              type="number"
              width={60}
              x={5}
              x1={59}
              x2={65}
              y={5}
              y1={215}
              y2={215}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={210}
              orientation="left"
              stroke="none"
              textAnchor="end"
              type="number"
              width={60}
              x={57}
              y={215}
            >
              <tspan
                dy="0.355em"
                x={57}
              >
                0
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={210}
              orientation="left"
              stroke="#666"
              type="number"
              width={60}
              x={5}
              x1={59}
              x2={65}
              y={5}
              y1={162.5}
              y2={162.5}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={210}
              orientation="left"
              stroke="none"
              textAnchor="end"
              type="number"
              width={60}
              x={57}
              y={162.5}
            >
              <tspan
                dy="0.355em"
                x={57}
              >
                50
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={210}
              orientation="left"
              stroke="#666"
              type="number"
              width={60}
              x={5}
              x1={59}
              x2={65}
              y={5}
              y1={110}
              y2={110}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={210}
              orientation="left"
              stroke="none"
              textAnchor="end"
              type="number"
              width={60}
              x={57}
              y={110}
            >
              <tspan
                dy="0.355em"
                x={57}
              >
                100
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={210}
              orientation="left"
              stroke="#666"
              type="number"
              width={60}
              x={5}
              x1={59}
              x2={65}
              y={5}
              y1={57.5}
              y2={57.5}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={210}
              orientation="left"
              stroke="none"
              textAnchor="end"
              type="number"
              width={60}
              x={57}
              y={57.5}
            >
              <tspan
                dy="0.355em"
                x={57}
              >
                150
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={210}
              orientation="left"
              stroke="#666"
              type="number"
              width={60}
              x={5}
              x1={59}
              x2={65}
              y={5}
              y1={5}
              y2={5}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={210}
              orientation="left"
              stroke="none"
              textAnchor="end"
              type="number"
              width={60}
              x={57}
              y={5}
            >
              <tspan
                dy="0.355em"
                x={57}
              >
                200
              </tspan>
            </text>
          </g>
        </g>
      </g>
      <g
        className="recharts-layer recharts-area"
      >
        <g
          className="recharts-layer recharts-area-dots"
          clipPath={null}
        >
          <circle
            className="recharts-dot recharts-area-dot"
            cx={280}
            cy={5}
            fill="#8884d8"
            fillOpacity={0.6}
            height={210}
            points={
              Array [
                Object {
                  "payload": Object {
                    "amount": 200,
                    "month": "jan",
                  },
                  "value": Array [
                    0,
                    200,
                  ],
                  "x": 280,
                  "y": 5,
                },
              ]
            }
            r={3}
            stroke="#8884d8"
            type="monotone"
            width={430}
          />
        </g>
      </g>
    </svg>
    <div
      className="recharts-tooltip-wrapper"
      style={
        Object {
          "MozTransform": "translate(undefinedpx, undefinedpx)",
          "OTransform": "translate(undefinedpx, undefinedpx)",
          "WebkitTransform": "translate(undefinedpx, undefinedpx)",
          "left": 0,
          "msTransform": "translate(undefinedpx, undefinedpx)",
          "pointerEvents": "none",
          "position": "absolute",
          "top": 0,
          "transform": "translate(undefinedpx, undefinedpx)",
          "visibility": "hidden",
        }
      }
    >
      <div
        className="recharts-default-tooltip"
        style={
          Object {
            "backgroundColor": "#fff",
            "border": "1px solid #ccc",
            "margin": 0,
            "padding": 10,
            "whiteSpace": "nowrap",
          }
        }
      >
        <p
          className="recharts-tooltip-label"
          style={
            Object {
              "margin": 0,
            }
          }
        >
          
        </p>
      </div>
    </div>
  </div>
</div>
`;
